contracts/commands/ModernTriggerExecutor.sol

```diff
@@ -0,0 +1,183 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+/// AaveStoplLossCommand.sol
+
+// Copyright (C) 2023 Oazo Apps Limited
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity 0.8.19;
+
+import { ICommand } from "../interfaces/ICommand.sol";
+import { IVerifier } from "../interfaces/IVerifier.sol";
+import { IServiceRegistry } from "../interfaces/IServiceRegistry.sol";
+import { IAccountImplementation } from "../interfaces/IAccountImplementation.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { IOperationExecutor, Call } from "../interfaces/Oasis/IOperationExecutor.sol";
+
+import { ReentrancyGuard } from "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+
+struct ModernTriggerData {
+    address positionAddress;
+    uint16 triggerType;
+    bytes32 operationHash;
+    // @BackToTheCode @robercano should we use a struct for this? or just 2 addresses? what if we want to have more tokens?
```

Comment:
If we expect this to have to handle more tokens, then yes, I'd use an array of addresses here. It actually matches the interface for `checkIfContractEmpty`, so it would fit nicely